runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerStepMap.java

```diff
@@ -42,27 +49,78 @@
  * <p>This class is not thread-safe.
  */
 public class MetricsContainerStepMap implements Serializable {
-  private Map<String, MetricsContainerImpl> metricsContainers;
+
+  /**
+   * Key class to lookup the metric container for a step or unbound to any step. This is preferable
+   * to using null(which can't be used as a key). Additonally, this class could be extended to allow
```

Comment:
Github doesn't seem to support a single "suggestion" block spanning multiple lines, but this causes a line-wrap so it's two "suggestion"s:
```suggestion
   * to using null (which can't be used as a key). Additionally, this class could be extended to
```